modules/api/pkg/ee/resource-quota/handler.go

```diff
@@ -172,35 +214,373 @@ func GetResourceQuota(ctx context.Context, request interface{}, provider provide
 func GetResourceQuotaForProject(ctx context.Context, request interface{}, projectProvider provider.ProjectProvider,
 	privilegedProjectProvider provider.PrivilegedProjectProvider, userInfoGetter provider.UserInfoGetter,
 	quotaProvider provider.ResourceQuotaProvider) (*apiv2.ResourceQuota, error) {
-	req, ok := request.(common.GetProjectRq)
+	projectResourceQuota, projectName, err := getProjectResourceQuota(ctx, request, projectProvider, privilegedProjectProvider, userInfoGetter, quotaProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	if projectResourceQuota == nil {
```

Comment:
From what I see, this code probably never executes. In case projectResourceQuota is not found kube returns 404 error (in the getProjectResourceQuota function)

Reply:
It does execute because of the way the getProjectResourceQuota works, specifically the resource quota provider:
https://github.com/kubermatic/dashboard/blob/03648c694cca330f62ad519cac4338a2bd9af6f5/modules/api/pkg/ee/resource-quota/provider.go#L98-L101

I am returning nil here because it seems to be an agreement between UI and API, as the GetResourceQuotaForProject worked the same for now. Probaby 404 is better, but I guess UI would then need to silence that error for projects which dont have resource quotas set

Reply:
Damn, you're actually right. In that case we need to leave it.